calculator.rb

```diff
@@ -0,0 +1,111 @@
+# If the result is a whole number, program should return an integer. Otherwise, it should return a float rounded to 2 decimal points.
+def checkForDecimal(result)
+  if result % 1 == 0
+    result = result.to_i
+  else
+    result = result.to_f.round(2)
+  end
+end
+
+# Methods for each operator
+def add(num1, num2)
+  result = num1 + num2
+  checkForDecimal(result)
+end
+
+def subtract(num1, num2)
+  result = num1 - num2
+  checkForDecimal(result)
+end
+
+def multiply(num1, num2)
+  result = num1 * num2
+  checkForDecimal(result)
+end
+
+def divide(num1, num2)
+  result = num1 / num2
+  checkForDecimal(result)
+end
+
+def exponent(num1, num2)
+  result = num1 ** num2
+  checkForDecimal(result)
+end
+
+def modulo(num1, num2)
+  result = num1 % num2
+  checkForDecimal(result)
+end
+
+# Methods for getting user numbers. Proram should keep prompting user until they enter an integer.
+def getFirstUserNum
+  begin
+    return Float(gets.chomp)
+  rescue
+    puts "Please enter the number as an integer."
+    retry
+  end
+end
+
+def getSecondUserNum
+  begin
+    return Float(gets.chomp)
+  rescue
```

Comment:
This method looks exactly the same as `getFirstUserNum`. This side-steps the big advantage that methods give us: we can use the same procedure to handle different data. Instead of defining two methods, you should be able to define one, maybe called `getUserNumber`, and use it in both places.